staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -115,7 +115,9 @@ func (c *controller) sync() {
 	}
 
 	lease, created := c.backoffEnsureLease()
-	c.latestLease = lease
+	if lease != nil {
+		c.latestLease = lease
+	}
```

Comment:
This doesn't seem like the right approach.

`backoffEnsureLease` will return a `nil` lease if none exists in the API **and** `newLease` returns an error (such as when the `Node` doesn't exist): https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go#L168

When that happens, if we *don't* set `c.latestLease` to `nil`, we'll enter the above `if` every time this function is called. `retryUpdateLease` will never succeed with your change below. We'll *then* call `ensureLease` here, with a chance of creating a new lease and stabilizing the state.

I think the approach in #119661 more directly addresses the issue, which is errors from `newLease` being ignored.